system-addon/lib/TopSitesFeed.jsm

```diff
@@ -84,18 +89,29 @@ this.TopSitesFeed = class TopSitesFeed {
       !NewTabUtils.blockedLinks.isBlocked({url: link.url}));
 
     // Get pinned links augmented with desired properties
-    const pinned = (await this.pinnedCache.request()).map(link => {
+    const plainPinned = await this.pinnedCache.request();
+    const pinned = await Promise.all(plainPinned.map(async link => {
       if (!link) {
         return link;
       }
 
       // Copy all properties from a frecent link and add more
       const finder = other => other.url === link.url;
-      return Object.assign({}, frecent.find(finder) || {}, link, {
+      const copy = Object.assign({}, frecent.find(finder) || {}, link, {
         hostname: shortURL(link),
         isDefault: !!notBlockedDefaultSites.find(finder)
       });
-    });
+
+      // Add in favicons if we don't already have it
+      if (!copy.favicon) {
+        NewTabUtils.activityStreamProvider._faviconBytesToDataURI(await
```

Comment:
Maybe just add a ```.catch``` to this ```Promise.all``` in case something goes wrong with one of these favicon fetching things, we don't want to reject all the others

Reply:
Or wrap in a try catch

Reply:
Ohh right. added `try/catch`